source/adios2/transform/compress/CompressZfp.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * CompressZfp.h : wrapper to Zfp compression library
+ *
+ *  Created on: Jul 25, 2017
+ *      Author: William F Godoy godoywf@ornl.gov
+ */
+
+#ifndef ADIOS2_TRANSFORM_COMPRESS_COMPRESSZFP_H_
+#define ADIOS2_TRANSFORM_COMPRESS_COMPRESSZFP_H_
+
+#include <zfp.h>
+
+#include "adios2/core/Transform.h"
+
+namespace adios2
+{
+
+namespace transform
+{
+
+class CompressZfp : public Transform
+{
+
+public:
+    /**
+     * Unique constructor
+     * @param debugMode
+     */
+    CompressZfp(const bool debugMode);
+
+    ~CompressZfp() = default;
+
+    /**
+     * Wrapper around zfp compression
+     * @param dataIn
+     * @param dimensions
+     * @param type
+     * @param bufferOut
+     * @param parameters
+     * @return size of compressed buffer in bytes
+     */
+    size_t Compress(const void *dataIn, const Dims &dimensions,
+                    const size_t elementSize, const std::string type,
+                    void *bufferOut, const Params &parameters) const final;
+
+    /**
+     * Wrapper around zfp decompression
+     * @param bufferIn
+     * @param sizeIn
+     * @param dataOut
+     * @param dimensions
+     * @param type
+     * @return size of decompressed data in dataOut
+     */
+    size_t Decompress(const void *bufferIn, const size_t sizeIn, void *dataOut,
+                      const Dims &dimensions, const std::string type,
+                      const Params &parameters) const final;
+
+private:
+    /**
```

Comment:
Are these helper functions using any data members?  If not then they can easily be made standalone instead of members and live entirely in the cpp file, out of the header.  That would also let you move the zfp include the cpp file to minimize include pollution in the interface.